lib/text_diff.ml

```diff
@@ -54,40 +54,138 @@ let pp_ diff_printer =
   in
   Fmt.list ~sep:Fmt.nop pp_dh
 
+let indent amount str =
+  let indentation = String.make amount ' ' in
+  indentation ^ str
+
+let indent_hunk amount hunk =
+  match hunk with
+  | Common lst -> Common (List.map (indent amount) lst)
+  | Line_conflict { orig; new_ } ->
+      Line_conflict
+        {
+          orig = List.map (indent amount) orig;
+          new_ = List.map (indent amount) new_;
+        }
+  | Inline_hunks ihunks ->
+      Inline_hunks
+        (match ihunks with
+        | [] -> ihunks
+        | ihunk :: ihunks' ->
+            (match ihunk with
+            | Icommon s -> Icommon (indent amount s)
+            | Iconflict { iorig; inew } ->
+                Iconflict
+                  {
+                    iorig = Option.map (indent amount) iorig;
+                    inew = Option.map (indent amount) inew;
+                  })
+            :: ihunks')
+
 let vd_to_lines name vd =
   let buf = Buffer.create 256 in
   let formatter = Format.formatter_of_buffer buf in
   Printtyp.value_description (Ident.create_local name) formatter vd;
   Format.pp_print_flush formatter ();
   CCString.lines (Buffer.contents buf)
 
-let td_to_lines name td =
-  let buf = Buffer.create 256 in
-  let formatter = Format.formatter_of_buffer buf in
-  Printtyp.type_declaration (Ident.create_local name) formatter td;
-  Format.pp_print_flush formatter ();
-  CCString.lines (Buffer.contents buf)
-
-let lbl_to_line ld =
-  let buf = Buffer.create 256 in
-  let formatter = Format.formatter_of_buffer buf in
-  Printtyp.label formatter ld;
-  Format.pp_print_flush formatter ();
-  String.map (function '\n' -> ' ' | c -> c) (Buffer.contents buf)
-
 let type_expr_to_string typ_exp =
   let buf = Buffer.create 256 in
   let formatter = Format.formatter_of_buffer buf in
   Printtyp.type_expr formatter typ_exp;
   Format.pp_print_flush formatter ();
   String.map (function '\n' -> ' ' | c -> c) (Buffer.contents buf)
 
-let cstr_to_lines ld =
-  let buf = Buffer.create 256 in
-  let formatter = Format.formatter_of_buffer buf in
-  Printtyp.constructor formatter ld;
-  Format.pp_print_flush formatter ();
-  CCString.lines ("| " ^ Buffer.contents buf)
+let params_to_string params =
+  let module TD = Intermed.TypeDecl in
+  match params with
+  | [] -> ""
+  | param :: [] -> " " ^ type_expr_to_string param.TD.type_expr
+  | _ ->
+      Printf.sprintf " (%s)"
+        (String.concat ", "
+           (List.map
+              (fun param -> type_expr_to_string param.TD.type_expr)
+              params))
+
+let private_to_string private_ = if private_ then " private" else ""
+let mutable_to_string mutable_ = if mutable_ then " mutable" else ""
+
+let field_to_string field =
+  let open Intermed.TypeDecl.Field in
+  let mutable_string = mutable_to_string field.mutable_ in
+  let type_expr_string = type_expr_to_string field.type_ in
+  Printf.sprintf " %s%s : %s;" mutable_string field.name type_expr_string
```

Comment:
There's someting a bit off there. If the field is mutable, then there will be two spaces at the start. I'm not sure this is intended.

The `to_string` functions should probably not handle surrounding spaces there as it otherwise becomes hard to track across the code. The caller should probably handle the spaces to separate the syntax elements.